src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -0,0 +1,244 @@
+                onPathFailure: reportingMode,
+                onLoaderFailure: reportingMode);
+
+            var projectPaths = ImmutableArray.CreateBuilder<string>();
+
+            // load all the projects
+            foreach (var project in solutionFile.ProjectsInOrder)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                if (project.ProjectType != MSB.Construction.SolutionProjectType.SolutionFolder)
+                {
+                    projectPaths.Add(project.RelativePath);
+                }
+            }
+
+            var worker = new Worker(
+                _workspace,
+                _diagnosticReporter,
+                _pathResolver,
+                _projectFileLoaderRegistry,
+                BuildManager,
+                projectPaths.ToImmutable(),
+                baseDirectory: Path.GetDirectoryName(absoluteSolutionPath),
+                _properties,
+                projectMap: null,
+                progress,
+                requestedProjectOptions,
+                discoveredProjectOptions,
+                preferMetadataForReferencedProjects: false);
+
+            var projects = await worker.LoadAsync(cancellationToken).ConfigureAwait(false);
+
+            // construct workspace from loaded project infos
+            return SolutionInfo.Create(
+                SolutionId.CreateNewId(debugName: absoluteSolutionPath),
+                version: default,
+                absoluteSolutionPath,
+                projects);
+        }
+
+        /// <summary>
+        /// Loads the <see cref="ProjectInfo"/> from the specified project file and all referenced projects.
+        /// The first <see cref="ProjectInfo"/> in the result corresponds to the specified project file.
+        /// </summary>
+        public async Task<ImmutableArray<ProjectInfo>> LoadProjectInfoAsync(
+            string projectFilePath,
+            ProjectMap projectMap = null,
+            IProgress<ProjectLoadProgress> progress = null,
+            CancellationToken cancellationToken = default)
+        {
+            if (projectFilePath == null)
+            {
+                throw new ArgumentNullException(nameof(projectFilePath));
+            }
+
+            var requestedProjectOptions = DiagnosticReportingOptions.ThrowForAll;
+
+            var reportingMode = GetReportingModeForUnrecognizedProjects();
+
+            var discoveredProjectOptions = new DiagnosticReportingOptions(
+                onPathFailure: reportingMode,
+                onLoaderFailure: reportingMode);
+
+            var worker = new Worker(
+                _workspace,
+                _diagnosticReporter,
+                _pathResolver,
+                _projectFileLoaderRegistry,
+                BuildManager,
+                requestedProjectPaths: ImmutableArray.Create(projectFilePath),
+                baseDirectory: Directory.GetCurrentDirectory(),
```

Comment:
>baseDirectory: Directory.GetCurrentDirectory() [](start = 16, length = 46)

Similar comment as GetCurrentDirectory above.

Reply:
See my comment on your other note. I've updated the doc comments to let the user know what values are legal and how they'll be used.